syntax = "proto2";

package sfdb;

import "google/protobuf/any.proto";
import "google/protobuf/descriptor.proto";

message ExecSqlRequest {
  optional string sql = 1;

  // Whether to include debug_strings in the response.
  optional bool include_debug_strings = 2;
}

message ExecSqlResponse {
  // Descriptors of included messages
  optional google.protobuf.FileDescriptorSet descriptors = 1;

  // Rows as dynamic messages
  repeated google.protobuf.Any rows = 2;

  // Rows as text-encoded protobufs (if include_debug_strings = true in request)
  repeated string debug_strings = 3;
}

service SfdbService {
  rpc ExecSql(ExecSqlRequest) returns (ExecSqlResponse);
}
